halo/cmd/ready.go

```diff
@@ -0,0 +1,61 @@
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/omni-network/omni/lib/errors"
+	"github.com/omni-network/omni/lib/log"
+
+	cmtcfg "github.com/cometbft/cometbft/config"
+
+	"github.com/spf13/cobra"
+)
+
+func newReadyCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "ready",
+		Short: "Assert the readiness of the halo node",
+		Args:  cobra.NoArgs,
+		RunE: func(cmd *cobra.Command, _ []string) error {
+			err := assertReady(cmd.Context())
+			if err != nil {
+				return errors.Wrap(err, "ready failed")
+			}
+
+			return nil
+		},
+	}
+
+	return cmd
+}
+
+// assertReady calls halo's /ready endpoint and returns nil if the status is ready
+// or an error otherwise.
+func assertReady(ctx context.Context) error {
```

Comment:
follow `status` command example, use `readyConfig` struct with a field: `MonitoringAddr` which defaults to `cmtconfig.DefaultInstrumentationConfig().PrometheusListenAddr`.